### PR TITLE
Locations bulk upload improvements

### PR DESCRIPTION
**Pull request by mkangia**
https://dimagi-dev.atlassian.net/browse/ICDS-1496

##### SUMMARY
Copying over from how we ensure the file uploaded is valid one for [user upload](https://github.com/dimagi/commcare-hq/blob/7cd8b855255a0a76c2e79d3920559fc6e64fff55/corehq/apps/users/views/mobile/users.py#L1017-L1021), to confirm both the file extension and format in a more reliable way for locations upload as well.
Also improved on what we show to the user in case of error during the import.

As for the code,
Looks like there is a need to make a common util that does this because this is done at multiple places. May be move the check to the bulk upload form `BulkUploadForm`? but leaving that for a later day

##### RISK ASSESSMENT / QA PLAN
I tested this myself locally and it worked as expected. A valid upload went through whereas the one with invalid file notified the user. Also check error message in case the task fails

##### PRODUCT DESCRIPTION
When doing a bulk locations upload,
Earlier the check was for pretty specific and lead to false errors.
The page now gives a valid error
when a doc file is uploaded
```
Invalid file-format. Please upload a valid xlsx file.
```

and when i upload a bad excel file
```
Upload failed! Please make sure you are using a valid Excel 2007 or later (.xlsx) file. Error details: File is not a zip file.
```
This is same as user's upload.

The "File is not a zip file" is confusing though, which i tried to see if we can resolve but it is coming from the third party excel library as the error message.

Also when there is an error in the upload, which happens during the import in celery task
The error shown was
`Problem importing data! Please try again or report an issue.`

I updated that for the user to know what exactly went wrong, so now it would be
```
Upload failed! Details:
'NoneType' object has no attribute 'has_permission'
```
(i made that to happen by failing the user lookup)

### CODE REVIEW

**Comment by mkangia** [comment]
@esoergel feel free to merge this if the build passes and changes look good, so that this gets in the next deploy

File: corehq/apps/locations/views.py [hunk 1]
```diff
@@ -932,10 +933,17 @@ def post(self, request, *args, **kwargs):
...
             return self.get(request, *args, **kwargs)
-        if upload.content_type != 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet':
+
+        if not upload.name.endswith('.xlsx'):
             messages.error(request, _("Invalid file-format. Please upload a valid xlsx file."))
             return self.get(request, *args, **kwargs)
 
+        try:
+            get_workbook(upload)
+        except WorkbookJSONError as e:
```
Comment (esoergel) [inline corehq/apps/locations/views.py:948]: :+1:

File: corehq/apps/locations/tasks.py [hunk 2]
```diff
@@ -100,35 +103,39 @@ def download_locations_async(domain, download_id, include_consumption, headers_o
...
-            " ".join(
-                "%s=%r" % (name, getattr(results, name))
-                for name in ["messages", "warnings", "errors"]
-                if getattr(results, name)
-            ),
-            results.success,
-        )
+    except Exception as e:
+        results = LocationUploadResult()
+        results.errors = [str(e)]
```
Comment (esoergel) [inline corehq/apps/locations/tasks.py:138]: Can you `notify_exception` here as well?  We'll probably want this in sentry.

**Review by esoergel — APPROVED** [review]

File: corehq/apps/locations/tasks.py (hunk 2 above)
Comment (mkangia) [reply corehq/apps/locations/tasks.py:138]: yikes! swallowed that,
https://github.com/dimagi/commcare-hq/pull/27530

**Pull request merged**